pkg/nsx/cluster.go

```diff
@@ -89,14 +93,65 @@ func (cluster *Cluster) NewRestConnector() (*policyclient.RestConnector, *Header
 }
 
 func (cluster *Cluster) createTransport(tokenProvider auth.TokenProvider, idle time.Duration) *Transport {
-	// TODO: support the case if InsecureSkipVerify is false
-	tlsConfig := tls.Config{InsecureSkipVerify: true}
+	dial := func(network, addr string) (net.Conn, error) {
+		host := strings.Split(addr, ":")[0]
+		var thumbprint string
+		tpCount := len(cluster.config.Thumbprint)
+		if tpCount == 1 {
+			thumbprint = cluster.config.Thumbprint[0]
+		}
+		if tpCount > 1 {
+			for index, ep := range cluster.endpoints {
+				if ep.Host() == host {
+					thumbprint = cluster.config.Thumbprint[index]
+					break
+				}
+			}
+		}
+		config := &tls.Config{
+			InsecureSkipVerify: true,
```

Comment:
Do we need to set `InsecureSkipVerify` as false when thumbprint is there

Reply:
no, even we verify the thumbprint, the InsecureSkipVerify is true, that verify here is provided by http lib, not include function we added.